### DISCUSSION TITLE
Cart Buyer Identity Fields: Associate customer address fields with cart

### DISCUSSION THREAD

**Discussion by nboliver** [Shopify/storefront-api-feedback#55] [Help]
With the checkout workflow, you can add the shipping address to the checkout via `checkoutShippingAddressUpdateV2`. This enables a user that has an access token to hit step 2 of the checkout immediately. 

We have moved to the Cart workflow in order to support subscriptions, but it appears the `CartBuyerIdentityInput` fields only support `email`, `phone`, `customerAccessToken`, and `countryCode`. When using the cart workflow, there should be a way to associate the current user's default address with the cart so they don't have to re-enter their info at the checkout. Perhaps there is already a way to do this, but it doesn't seem like it from the docs. 

(I categorized this discussion as Help, but if there is no way to do this, I'll change it to Ideas and Requests)

**Comment by aurelieyeh** [comment]
Hey @nboliver , our team is looking into this. We'll get back to you shortly!

**Reply by patjakubik** [reply]
Hey @aurelieyeh Any updates on this?

**Comment by aurelieyeh** [comment]


**Reply by nboliver** [reply]
Thanks @aurelieyeh. That link is 404 for me. Is it private?

**Reply by Francismori7** [reply]
> Thanks @aurelieyeh. That link is 404 for me. Is it private?

Yes, it's private!

**Reply by nboliver** [reply]
@aurelieyeh @Francismori7 Any update here?

**Comment by nboliver** [comment]
@vixdug Maybe you can chime in? We have gone ahead and launched an app with a hack to pass customer data to checkout via query string and pre-fill the form fields, but this is not a great solution long term. Is this functionality on the roadmap somewhere?

**Reply by behappyenjoylife** [reply]
@nboliver can you please share that hack with us, as we are facing same issue that on checkout we are not able to pre-fill the details of customer on the checkout page

**Reply by NIHAL-JAIN** [reply]
@behappyenjoylife 
Best way you can breakdown checkout into 3 pages 
<img width="1280" alt="Screenshot 2024-06-18 at 1 49 55 PM" src="https://github.com/Shopify/storefront-api-feedback/assets/55236143/d34e1651-db61-43ce-9812-8f2f0d1fba9a">

But still user is not login you can check below SS, only we are moving to shipping page without filling info, with this approach we cannot get customer default address

<img width="1280" alt="Screenshot 2024-06-19 at 3 37 50 PM" src="https://github.com/Shopify/storefront-api-feedback/assets/55236143/05ae6305-24c2-4a4c-8036-24a57f067dbf">

<img width="1280" alt="Screenshot 2024-06-19 at 3 38 33 PM" src="https://github.com/Shopify/storefront-api-feedback/assets/55236143/c984bf85-594d-405b-9c9a-40bdcec8170c">

**Reply by NIHAL-JAIN** [reply]
Sorry @behappyenjoylife but with this we cannot select default address of user, it's a stopper for me every time user cannot type address we have to show them dropdown of saved address

**Reply by nboliver** [reply]
I believe the API supports pre-filling now. https://shopify.dev/changelog/new-preferences-fields-for-prefilling-checkout-in-the-storefront-cart-api
It only took two years to happen 😂

**Reply by NIHAL-JAIN** [reply]
if i don't want guest checkout then how it will work and how we are taking user default address as well

**Reply by NIHAL-JAIN** [reply]
@nboliver if you have any solution please let me know, I was stuck here for more then a week
